app/Main.hs

```diff
@@ -66,11 +67,16 @@ main = do
 initializeAllFeatures :: CLIArguments -> PartialCardanoConfiguration -> CardanoEnvironment -> IO ([CardanoFeature], NodeLayer)
 initializeAllFeatures (CLIArguments logCli nodeCli) partialConfig cardanoEnvironment = do
 
+    let pccCore' = pccCore partialConfig :: Last PartialCore
+
     finalConfig <- either (throwIO . ConfigurationError) pure $
           finaliseCardanoConfiguration $
-          -- Here we perform merging of layers of configuration, but for now, only in a trivial way.
+          -- Here we perform merging of layers of configuration, but for now, only in a trivial way,
+          -- just for Cardano.Shell.Constants.Types.Genesis.
           -- We expect this process to become generic at some point.
-          partialConfig { pccNode = pccNode partialConfig }
+          partialConfig { pccCore = pccCore' <> (pure $ mempty { pcoGenesis = (join $ pcoGenesis <$> pccCore')
+                                                                 <> genesisSpec nodeCli
+                                                               })}
```

Comment:
```
pccCore' <> pure (...)
```
This is wrong, or not what you meant. Think about it. You've said that `pccCore' :: Last ...`, and now you're using the monoid operation with a pure (i.e. `Last`) as the right hand arg.

For `Last`:
```
forall x.  x <> pure y = pure y
```

Also, this expression is getting unwieldy. You've tried to make it smaller using a `let` but put the `let` far away. You can use a local let and make this all fit in 80 cols. Note also that we can probably simplify the merging of the config if the top level thing is not a `Last` monoid but a product monoid. e.g. something like:
```
finaliseCardanoConfiguration $
-- the comments
let mergeCLI :: PartialCore -> PartialCore
    mergeCLI core = core <> mempty { pcoGenesis = genesisSpec nodeCli }
in partialConfig { pccCore = fmap mergeCLI (pccCore partialConfig) }
```

Reply:
> This is wrong, or not what you meant. Think about it. You've said that `pccCore' :: Last ...`, and now you're using the monoid operation with a pure (i.e. `Last`) as the right hand arg.

Indeed.. I wanted a component-wise monoidal append, but got a whole-sale override by the right-side in its stead.  Thank you for catching it!

Reply:
Would you discourage a point-free variant I submitted?

Reply:
The rabbit hole was embarassingly uniform.  The nested children need the same treatment.

Reply:
Yes point-free is discouraged, (which includes over using of `Control.Arrow`), often it's much easier to read non point-free, with explicit lambdas.  But I haven't seen your point-free term, so take it as guide-line, which at times might be good to break.

Reply:
Yes, the point-free variant did not survive in the end -- the whole thing was isolated into a separate function: https://github.com/input-output-hk/cardano-node/pull/60/files#diff-ba9f0f06843a3a67a21c656d60e82bc8R175